src/Events/Controllers/CollectController.cs

```diff
@@ -77,7 +77,15 @@ public async Task<IActionResult> Post([FromBody] IEnumerable<EventModel> model)
                     }
                     if (cipher == null)
                     {
-                        continue;
+                        // When the user cannot access the cipher directly, check if the organization allows for
+                        // admin/owners access to all collections and the user can access the cipher from that perspective.
+                        var org = await _organizationRepository.GetByIdAsync(eventModel.OrganizationId.Value);
+                        cipher = await _cipherRepository.GetByIdAsync(eventModel.CipherId.Value);
+
+                        if (!org.AllowAdminAccessToAllCollectionItems || cipher == null)
```

Comment:
⚠️ We don't actually know if the current context User belongs to the `eventModel.OrganizationId` and is an Admin to know if the `AllowAdminAccessToAllCollectionItems` settings applies to them. The `_currentContext` has a method called `GetOrganization(orgId)` that will return the org if the user is a member in some capacity. It returns an object that we can check the user `Type` of to be sure they're and Admin/Owner. 

Note: We **cannot** use the `_currentContext.OrganizationAdmin()` helper because it will return `true` for providers who should *not* have access to organization ciphers.

A good example to reference is here: https://github.com/bitwarden/server/blob/0a6c35e56c0e0ffa4368ec81fbfb0eb9eeb9cd8d/src/Api/Vault/Controllers/CiphersController.cs#L352-L367

Reply:
Great context, thanks! I believe I have addressed the both this comment and your [above comment](https://github.com/bitwarden/server/pull/4705#discussion_r1733239111) following the same pattern you linked here. Let me know if I missed anything!

[7c2d18a](https://github.com/bitwarden/server/pull/4705/commits/7c2d18ae7af58fc217c1aab38295e529d546b8d8)